### PR TITLE
feat(log): Add possibility to enable application debug through UI

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to Add possibility to enable application debug through UI

**Fixes** # MOn-15141

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by arnapou — DISMISSED** [review]

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/help.po [hunk 1]
```diff
@@ -7334,3 +7334,9 @@ msgstr "Coordonnées géographiques permettant à Centreon MAP de positionner le
 
 msgid "Provided that the user can access the API Tokens menu, this action will allow them to not only list, create and delete tokens for themselves, but also manage tokens for other users."
 msgstr "Si l'utilisateur peut accéder au menu des jetons API, cette action lui permettra non seulement de répertorier, de créer et de supprimer des jetons pour lui-même, mais aussi de gérer les jetons d'autres utilisateurs."
+
+msgid "Enables application logs located in /var/log/centreon/centreon-web.log"
+msgstr "Permet d'activer les logs applicatif situés dans /var/log/centreon/centreon-web.log"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/help.po:7339]:
```suggestion
msgstr "Permet d'activer les logs applicatifs situés dans /var/log/centreon/centreon-web.log"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/help.po [hunk 2]
```diff
@@ -7334,3 +7334,9 @@ msgstr "Coordonnées géographiques permettant à Centreon MAP de positionner le
...
+
+msgid "Set the lowest log level: Debug => Info => Notice => Warning => Error => Critical => Alert => Emergency"
+msgstr "Définis le niveau de log le plus bas: Debug => Info => Notice => Warning => Error => Critical => Alert => Emergency"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/help.po:7342]:
```suggestion
msgstr "Définit le niveau de log le plus bas: Debug => Info => Notice => Warning => Error => Critical => Alert => Emergency"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -18329,3 +18329,6 @@ msgstr "Erreur lors de la mise à jour d'un hôte"
 
 msgid "You are not allowed to edit a host"
 msgstr "Vous n'êtes pas autorisé à éditer un hôte"
+
+msgid "Application debug"
+msgstr "Debug Applicatif"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18334]:
```suggestion
msgstr "Debug applicatif"
```

File: centreon/www/include/Administration/parameters/debug/form.ihtml [hunk 4]
```diff
@@ -13,6 +13,8 @@
     </tr>
 	<tr class="list_one"><td class="FormRowField"><img class="helpTooltip" name="tip_logs_directory">{$form.debug_path.label}</td><td class="FormRowValue">{$form.debug_path.html}</td></tr>
 	<tr class="list_two"><td class="FormRowField"><img class="helpTooltip" name="tip_authentication_debug">{$form.debug_auth.label}</td><td class="FormRowValue">{$form.debug_auth.html}</td></tr>
+	<tr class="list_two"><td class="FormRowField"><img class="helpTooltip" name="tip_application_debug">{$form.debug_application.label}</td><td class="FormRowValue">{$form.debug_application.html}</td></tr>
+	<tr class="list_two"><td class="FormRowField"><img class="helpTooltip" name="tip_debug_level">{$form.debug_level.label}</td><td class="FormRowValue">{$form.debug_level.html}</td></tr>
```
Comment (jeremyjaouen) [inline centreon/www/include/Administration/parameters/debug/form.ihtml:17]: Remove the checkbox and only handle the level listing with a None for debug deactivated

Comment (jeremyjaouen) [inline centreon/www/include/Administration/parameters/debug/form.ihtml:17]:
```suggestion
	<tr class="list_two"><td class="FormRowField"><img class="helpTooltip" name="tip_debug_level">{$form.debug_level.label}</td><td class="FormRowValue">{$form.debug_level.html}</td></tr>
```

**Review by adr-mo — APPROVED** [review]

File: centreon/www/include/Administration/parameters/debug/form.php [hunk 5]
```diff
@@ -55,6 +55,23 @@
 $form->addElement('text', 'debug_path', _("Logs Directory"), $attrsText);
 
 $form->addElement('checkbox', 'debug_auth', _("Authentication debug"));
+$form->addElement('checkbox', 'debug_application', _("Application debug"));
```
Comment (arnapou) [inline centreon/www/include/Administration/parameters/debug/form.php:58]: This line is superfluous, isn't it ? 
```suggestion
```

**Review by arnapou — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Review by arnapou — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**